### PR TITLE
Code reorganization

### PR DESCRIPTION
**Pull request by SergiiDmytruk**
This PR is on top of #46, will rebase on doing changes there.

This is a preparation of existing code for introducing TPM2 support in #42, it contains:
 * several small changes
 * a bunch of commits that extract TPM-dependent functionality to `anti-evil-maid-lib` as functions
 * move of TPM-related functions to `anti-evil-maid-lib-tpm1` file

My tests on TPM1 without SRK password, with SRK password and with TOTP went without errors.

### CODE REVIEW

File: sbin/anti-evil-maid-install [hunk 1]
```diff
@@ -62,11 +63,12 @@ END
 # check invocation
 
 alias mfa=false
-unset LABEL_SUFFIX F
+LABEL_SUFFIX=
+F=()
```
Comment (marmarek) [inline sbin/anti-evil-maid-install:67]: Array are bash feature, please change shebang when using it.

File: 90anti-evil-maid/anti-evil-maid-unseal [hunk 2]
```diff
@@ -58,6 +58,8 @@ fi
 log "Mounting $DEV (\"$LABEL\")..."
 mkdir -p "$MNT"
 mount -t ext4 -o ro "$DEV" "$MNT"
+# this way an error prior to unmounting will keep the device usable after initrd
+trap 'umount --quiet "$MNT" || true' EXIT
```
Comment (marmarek) [inline 90anti-evil-maid/anti-evil-maid-unseal]: The `|| true` will hide other errors too, not only the "not mounted" one. Maybe clear the trap when unmounting explicitly later in this script instead?

File: sbin/anti-evil-maid-lib-tpm1 [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+    # extends a PCR with a hash value of a suitable type
+    _pcr=$1
+    _hash=$2
+    tpm_pcr_extend "$_pcr" "$_hash"
+}
+
+tpmsealdata() {
+    # seals source specified by second argument into destination specified by
+    # the third one, non-empty first argument signifies empty SRK password
+    _nosrkpass=()
```
Comment (marmarek) [inline sbin/anti-evil-maid-lib-tpm1:162]: similar as in the other place - arrays are bash feature, ensure that all scripts that source this lib have bash shebang (currently some have just `#!/bin/sh`)

**Review by marmarek — CHANGES_REQUESTED** [review]

File: sbin/anti-evil-maid-lib-tpm1 (hunk 3 above)
Comment (SergiiDmytruk) [reply sbin/anti-evil-maid-lib-tpm1:162]: Except for `sbin/anti-evil-maid-install` the only other script with `/bin/sh` is `etc/grub.d/19_linux_xen_tboot` and it doesn't source `anti-evil-maid-lib`.

File: sbin/anti-evil-maid-lib-tpm1 [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+    # create a write-once NVRAM area
+    tpm_nvdefine -i "$_tpm_id_index" -s 20 -p "WRITEDEFINE|WRITEALL" \
+                 --pwdo="$_opw"
+    # generate a random ID and write it into NVRAM
+    head -c 20 /dev/random | tpm_nvwrite_stdin -i "$_tpm_id_index" -s 20
+    # lock the area to prevent non-owners from changing ID
+    tpm_nvwrite -i "$_tpm_id_index" -s 0
+}
+
+postprovisioning() {
```
Comment (krystian-hebel) [inline sbin/anti-evil-maid-lib-tpm1:135]: Is this specific to the TPM family? I haven't seen any significant changes touching `$TPMS_DIR` in TPM2.0 code, but maybe I missed it.

Comment (SergiiDmytruk) [reply sbin/anti-evil-maid-lib-tpm1:135]: TPM1 needs to create TPM-specific directory, deal with `tcsd` and move files.  TPM2 only creates the directory.

Comment (krystian-hebel) [reply sbin/anti-evil-maid-lib-tpm1:135]: I see.

**Review by krystian-hebel — APPROVED** [review]

**Review by marmarek — APPROVED** [review]

**Pull request merged**